app/src/main/java/com/kkolontay/theworld/ui/screens/WorldNavigation.kt

```diff
@@ -27,70 +30,60 @@ enum class AppScreens() {
+        mutableStateOf("")
+    }
+    var choosenCountry = remember {
+        mutableStateOf<Country>(Country(name = CountryName(common = "some"), capital = listOf("other"), population = 34, area = 45.0, flags = CountryFlags(png = "some")))
     }
-//    var choosenCountry by remember {
-//        mutableStateOf<Country>(Country(name = CountryName(common = "some"), capital = listOf("other"), population = 34, area = 45.0, flags = CountryFlags(
-//            png = "some"
-//        )))
-//    }
     Scaffold(
         topBar = {
             AppBar(
                 title = title.value,
                 canNavigateBack = navController.previousBackStackEntry != null,
                 navigateUp = {
                     title.value = context.getString(R.string.list1)
+                    MainScope().launch {
+                        viewModel.flows.tapBack()
+                    }
                     navController.navigateUp()
                 }
             )
         }
     ) { innerPadding ->
-        val countries = remember {
-            listOf(
-            Country(name = CountryName(common = "some"), capital = listOf("other"), population = 34, area = 45.0, flags = CountryFlags(
-            png = "https://mainfacts.com/media/images/coats_of_arms/md.png"
-        )
-            ),
-            Country(name = CountryName(common = "some"), capital = listOf("other"), population = 34, area = 45.0, flags = CountryFlags(
-            png = "https://mainfacts.com/media/images/coats_of_arms/md.png"
-        )
-            ), Country(name = CountryName(common = "some"), capital = listOf("other"), population = 34, area = 45.0, flags = CountryFlags(
-            png = "https://mainfacts.com/media/images/coats_of_arms/md.png"
-        )
-            ), Country(name = CountryName(common = "some"), capital = listOf("other"), population = 34, area = 45.0, flags = CountryFlags(
-            png = "https://mainfacts.com/media/images/coats_of_arms/md.png"
-        )
-            ), Country(name = CountryName(common = "some"), capital = listOf("other"), population = 34, area = 45.0, flags = CountryFlags(
-            png = "https://mainfacts.com/media/images/coats_of_arms/md.png"
-        )
-            )
-        )
-        }
+        val uiState = remember {viewModel.uiState}
+        var timerState = viewModel.timerState.collectAsState()
+        var  tap = viewModel.tap.collectAsState()
+        var back = viewModel.back.collectAsState()
 
-        //val uiState by viewModel.uiState.collectAsState()
         NavHost(navController = navController, startDestination = AppScreens.ListCountry.name, modifier = Modifier.padding(innerPadding)) {
             composable(AppScreens.ListCountry.name) {
-                CountryList(countries = countries) {
-                   // choosenCountry.value = it
+                CountryList(countries = uiState.collectAsState().value, timer = timerState.value, taps = tap.value, back = back.value, refresh = {
```

Comment:
Rather than pulling the referenced values from the viewmodel out and setting them into CountryList, you could instead pass the whole viewmodel into CountryList.  This encapsulates the logic inside CountryList, while also making this callsite simpler:

```kotlin
CountryList(viewModel = viewModel, refresh = {
    ...
}) { ...
```